test/extended/kubevirt/util.go

```diff
@@ -353,23 +363,119 @@ func InKubeVirtClusterContext(oc *exutil.CLI, body func()) {
 
 	return oc
 }
 
+func virtualMachineInstanceClient(dc dynamic.Interface) dynamic.NamespaceableResourceInterface {
+	return dc.Resource(vmiGVR)
+}
+
+func virtualMachineInstanceMigrationClient(dc dynamic.Interface) dynamic.NamespaceableResourceInterface {
+	return dc.Resource(vmimGVR)
+}
+
+func composeVMIM(vmiNamespace, vmiName string) *unstructured.Unstructured {
+	vmim := unstructured.Unstructured{}
+	vmim.SetGroupVersionKind(vmimGVK)
+	vmim.SetNamespace(vmiNamespace)
+	vmim.SetGenerateName(vmiName + "-")
+	vmim.UnstructuredContent()["spec"] = map[string]interface{}{
+		"vmiName": vmiName,
+	}
+	return &vmim
+}
+
+func migrateWorkers(f *e2e.Framework) error {
+	By("migrating hosted cluster workers")
+	vmiClient := virtualMachineInstanceClient(f.DynamicClient)
+	vmiList, err := vmiClient.Namespace(f.Namespace.Name).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed retrieving workers VMs at namespace %s: %v", f.Namespace.Name, err)
+	}
+
+	vmimClient := virtualMachineInstanceMigrationClient(f.DynamicClient)
+	vmims := []*unstructured.Unstructured{}
+	for _, vmi := range vmiList.Items {
+		vmim := composeVMIM(vmi.GetNamespace(), vmi.GetName())
+		if vmim, err = vmimClient.Namespace(f.Namespace.Name).Create(context.Background(), vmim, metav1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed creating vmim %s/%s: %v", vmi.GetNamespace(), vmi.GetName(), err)
+		}
+		vmims = append(vmims, vmim)
+	}
+	for _, vmim := range vmims {
+		phase := ""
+		err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+			vmim, err = vmimClient.Namespace(f.Namespace.Name).Get(context.Background(), vmim.GetName(), metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			statusIface, ok := vmim.UnstructuredContent()["status"]
+			if !ok {
+				fmt.Println("deleteme 1")
+				return false, nil
+			}
+			status, ok := statusIface.(map[string]interface{})
+			if !ok {
+				return false, fmt.Errorf("bad vmim.status")
+			}
+			phaseIface, ok := status["phase"]
+			if !ok {
+				fmt.Println("deleteme 2")
+				return false, nil
+			}
+			phase, ok = phaseIface.(string)
+			if !ok {
+				return false, fmt.Errorf("bad vmim.status.phase")
+			}
+			return phase == "Succeeded" || phase == "Failed", nil
+		})
+		if err != nil {
+			return err
+		}
+		if phase != "Succeeded" {
+			return fmt.Errorf("migration failed")
+		}
+	}
+	return nil
+}
+
 func expectNoError(err error, explain ...interface{}) {
 	ExpectWithOffset(1, err).NotTo(HaveOccurred(), explain...)
 }
+
+func mgmtClusterIsAWS(f *e2e.Framework) (bool, error) {
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+	if len(nodeList.Items) == 0 {
+		return false, fmt.Errorf("missing mgmt cluster nodes")
+	}
+	firstNode := nodeList.Items[0]
+	return strings.HasPrefix(firstNode.Spec.ProviderID, "aws:"), nil
+}
```

Comment:
You can look at the infrastructure object on the mgmt cluster to get whether it is aws or not.

example using cli... i'm pretty sure there are lots of examples in the test suite of this as well. 
```
$ oc get infrastructure -o yaml | grep "platform\:"
```